src/mds/QuiesceDbManager.cc

```diff
@@ -74,8 +74,24 @@ void* QuiesceDbManager::quiesce_db_thread_main()
 
     auto db_age = db.get_age();
 
-    if (!db_thread_has_work() && next_event_at_age > db_age) {
-      submit_condition.wait_for(ls, next_event_at_age - db_age);
+    // QuiesceTimeInterval::max() may overflow inside the call to
+    // std::condition_variable::wait_for and result in a busy-loop,
+    // making the call to timeout immediately
```

Comment:
I think more precisely you could say: `next_event_at_age - db_age` may exceed the size of the timeout variable used in the syscall.